To: vim_dev@googlegroups.com
Subject: Patch 7.3.144
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.144
Problem:    Crash with ":python help(dir)". (Kearn Holliday)
Solution:   Fix the way the type is set on objects. (Tobias Columbus)
Files:	    src/if_python.c


*** ../vim-7.3.143/src/if_python.c	2010-11-16 19:25:56.000000000 +0100
--- src/if_python.c	2011-03-26 13:54:22.000000000 +0100
***************
*** 780,786 ****
  PythonIO_Init(void)
  {
      /* Fixups... */
!     OutputType.ob_type = &PyType_Type;
  
      return PythonIO_Init_io();
  }
--- 780,786 ----
  PythonIO_Init(void)
  {
      /* Fixups... */
!     PyType_Ready(&OutputType);
  
      return PythonIO_Init_io();
  }
***************
*** 1402,1413 ****
      static char *(argv[2]) = {"/must>not&exist/foo", NULL};
  
      /* Fixups... */
!     BufferType.ob_type = &PyType_Type;
!     RangeType.ob_type = &PyType_Type;
!     WindowType.ob_type = &PyType_Type;
!     BufListType.ob_type = &PyType_Type;
!     WinListType.ob_type = &PyType_Type;
!     CurrentType.ob_type = &PyType_Type;
  
      /* Set sys.argv[] to avoid a crash in warn(). */
      PySys_SetArgv(1, argv);
--- 1402,1413 ----
      static char *(argv[2]) = {"/must>not&exist/foo", NULL};
  
      /* Fixups... */
!     PyType_Ready(&BufferType);
!     PyType_Ready(&RangeType);
!     PyType_Ready(&WindowType);
!     PyType_Ready(&BufListType);
!     PyType_Ready(&WinListType);
!     PyType_Ready(&CurrentType);
  
      /* Set sys.argv[] to avoid a crash in warn(). */
      PySys_SetArgv(1, argv);
*** ../vim-7.3.143/src/version.c	2011-03-22 18:10:34.000000000 +0100
--- src/version.c	2011-03-26 13:56:15.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     144,
  /**/

-- 
The chat program is in public domain.  This is not the GNU public license.
If it breaks then you get to keep both pieces.
		-- Copyright notice for the chat program

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
